src/widget/woverviewrgb.cpp

```diff
@@ -71,24 +84,36 @@ bool WOverviewRGB::drawNextPixmapPart() {
         unsigned char left = pWaveform->getAll(currentCompletion);
         unsigned char right = pWaveform->getAll(currentCompletion + 1);
 
-        low = pWaveform->getLow(currentCompletion);
-        mid = pWaveform->getMid(currentCompletion);
-        high = pWaveform->getHigh(currentCompletion);
+        low  = static_cast<qreal>(pWaveform->getLow(currentCompletion));
+        mid  = static_cast<qreal>(pWaveform->getMid(currentCompletion));
+        high = static_cast<qreal>(pWaveform->getHigh(currentCompletion));
+
+        // Do matrix multiplication
+        red   = low * lowColor_r + mid * midColor_r + high * highColor_r;
+        green = low * lowColor_g + mid * midColor_g + high * highColor_g;
+        blue  = low * lowColor_b + mid * midColor_b + high * highColor_b;
 
-        max = (float) math_max3(low, mid, high);
-        if (max > 0.0f) {
-            color.setRgbF(low / max, mid / max, high / max);
+        // Normalize and draw
+        max = math_max3(red, green, blue);
+        if (max > 0.0) {
+            color.setRgbF(red / max, green / max, blue / max);
             painter.setPen(color);
             painter.drawLine(currentCompletion / 2, -left, currentCompletion / 2, 0);
         }
 
-        low = pWaveform->getLow(currentCompletion + 1);
-        mid = pWaveform->getMid(currentCompletion + 1);
-        high = pWaveform->getHigh(currentCompletion + 1);
+        low  = static_cast<qreal>(pWaveform->getLow(currentCompletion + 1));
```

Comment:
Btw, I noticed a similar problem in https://github.com/mixxxdj/mixxx/blob/master/src/waveform/renderers/glwaveformrendererrgb.cpp#L105. Do you want me to fix this as well?